src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,24 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_RESIDENCE_TIME = 10;
+    private static final int MIN_AGE = 35;
+    private static final String NATIONALITY = "Ukrainian";
+
+    private boolean rightOfResidence(String years) {
```

Comment:
```suggestion
    private boolean hasRightOfResidence(String years) {
```
let's also move this method to the end of class